src/common/utility_win.cpp

```diff
@@ -293,4 +295,31 @@ QString Utility::formatWinError(long errorCode)
     return QStringLiteral("WindowsError: %1: %2").arg(QString::number(errorCode, 16), QString::fromWCharArray(_com_error(errorCode).ErrorMessage()));
 }
 
+
+Utility::NtfsPermissionLookupRAII::NtfsPermissionLookupRAII()
+{
+    enable();
+}
+
+Utility::NtfsPermissionLookupRAII::~NtfsPermissionLookupRAII()
+{
+    disable();
+}
+
+void Utility::NtfsPermissionLookupRAII::enable()
+{
+    if (!_enabled) {
+        _enabled = true;
+        qt_ntfs_permission_lookup++;
+    }
+}
+
+void Utility::NtfsPermissionLookupRAII::disable()
+{
+    if (_enabled) {
+        _enabled = false;
+        qt_ntfs_permission_lookup--;
+    }
+}
```

Comment:
Wouldn't it be enough to simply increment/decrement `qt_ntfs_permission_lookup` in the ctor/dtor?

What is _enabled good for?

And without that do we really need enable and disable methods? Looks kinda verbose ... but do whatever you like

Reply:
the boolean is to prevent negative values for qt_ntfs_permission_lookup as he code there relally just checks for  qt_ntfs_permission_lookup > 0

Reply:
Yes, that's what I'm saying:

```
Utility::NtfsPermissionLookupRAII::NtfsPermissionLookupRAII()
{
   qt_ntfs_permission_lookup++;
}
Utility::NtfsPermissionLookupRAII::~NtfsPermissionLookupRAII()
{
   qt_ntfs_permission_lookup--;
}
```

... should be enough.